src/main/java/htsjdk/samtools/SamFileValidator.java

```diff
@@ -301,7 +311,10 @@ private void validateSamRecordsAndQualityFormat(final Iterable<SAMRecord> samRec
                     }
                 }
 
-                validateMateFields(record, recordNumber);
+                if (!skipMateValidation) {
+                    validateMateFields(record, recordNumber);
```

Comment:
this skips the mate-cigar validation as well, would it be enough to move the if statement into validateMateFields and avoid the expensive storing of the reads, but not the validation of the mate-cigars?